File: Data Structures & Algorithms/counting-bits/submission-5.java
class Solution {
    public int[] countBits(int n) {
        int[] res = new int[n + 1];

        for (int i = 0; i <= n; i++) {
            int counter = 0;
            int num = i;

            for (int j = 0; j < 32; j++) {
                if ((num & 1) == 1) {
                    counter += 1;
                }
                num >>= 1;
            }

            res[i] = counter;
        }

        return res;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.py
from collections import Counter 
import heapq 
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        #bucket sort method for o(n) time 
        res = [] 
        counter = Counter(nums)
        count = [[] for i in range(len(nums) + 1)]

        for num,freq in counter.items():
            count[freq].append(num) 

        for i in range(len(count) - 1, 0, -1):
            for num in count[i]:
                res.append(num)

            if len(res) == k:
                return res 
